packages/language-service/src/typescript_host.ts

```
@@ -175,8 +177,13 @@ export class TypeScriptServiceHost implements LanguageServiceHost {
       }
     };
     const programFiles = this.program.getSourceFiles().map(sf => sf.fileName);
-    this.analyzedModules =
-        analyzeNgModules(programFiles, analyzeHost, this.staticSymbolResolver, this.resolver);
+
+    try {
+      this.analyzedModules = this.analyzeNgModules(
+          programFiles, analyzeHost, this.staticSymbolResolver, this.resolver);
+    } catch {
+      // Analyzing modules may throw; in that case, reuse the old modules.
+    }
```

We should propagate the error to an end user, maybe by attaching a logger to the host.

---

The logger method is right in this class

https://github.com/angular/angular/blob/7acd33007d9bd30288a76fa2a3b6fe5948143d9f/packages/language-service/src/typescript_host.ts#L557-L566